libnmstate/ifaces/ethernet.py

```diff
@@ -79,6 +78,20 @@ def is_sriov(self):
             Ethernet.SRIOV_SUBTREE
         )
 
+    @property
+    def speed(self):
+        return self.raw.get(Ethernet.CONFIG_SUBTREE, {}).get(Ethernet.SPEED)
+
+    @property
+    def auto_negotiation(self):
+        return self.raw.get(Ethernet.CONFIG_SUBTREE, {}).get(
+            Ethernet.AUTO_NEGOTIATION
+        )
+
+    @property
+    def duplex(self):
```

Comment:
Please, could you add unit tests covering the new properties? Thanks!

Reply:
Updated existing unit test case to cover these new properties.